services/src/main/java/org/exoplatform/officeonline/WOPIService.java

```diff
@@ -795,4 +807,173 @@ public void refreshLock(EditorConfig config,
   }
 
 
+  /**
+   * Put relative file in suggested mode.
+   *
+   * @param config the config
+   * @param target the target
+   * @param data the data
+   * @throws RepositoryException the repository exception
+   * @throws FileNotFoundException the file not found exception
+   * @throws FileExtensionNotFoundException 
+   */
+  public void putSuggestedFile(EditorConfig config, String target, InputStream data) throws RepositoryException,
+                                                                                     FileNotFoundException,
+                                                                                     FileExtensionNotFoundException {
+    Node node = nodeByUUID(config.getFileId(), config.getWorkspace());
+    String oldName = node.getName();
+    String filename = target;
+    // Target is an extension
+    if (target.startsWith(".")) {
+      String name = oldName.substring(0, oldName.lastIndexOf("."));
+      filename = name + target;
+    }
+
+    filename = Text.escapeIllegalJcrChars(filename);
+    // Check and escape newTitle
+    if (StringUtils.isBlank(filename)) {
+      filename = DEFAULT_FILENAME + oldName.substring(oldName.lastIndexOf("."));
+    }
+
+    Node parent = node.getParent();
+    createFile(parent, filename, data);
+  }
+
+  /**
+   * Put relative file.
+   *
+   * @param config the config
+   * @param filename the filename
+   * @param overwrite the overwrite
+   * @param data the data
+   * @throws IllegalFileNameException the illegal file name exception
+   * @throws FileNotFoundException the file not found exception
+   * @throws RepositoryException the repository exception
+   * @throws FileLockedException the file locked exception
+   * @throws UpdateConflictException the update conflict exception
+   * @throws FileExtensionNotFoundException the file extension not found exception
+   */
+  public void putRelativeFile(EditorConfig config,
+                              String filename,
+                              boolean overwrite,
+                              InputStream data) throws IllegalFileNameException,
+                                                FileNotFoundException,
+                                                RepositoryException,
+                                                FileLockedException,
+                                                UpdateConflictException,
+                                                FileExtensionNotFoundException {
+    filename = Text.escapeIllegalJcrChars(filename);
+    // Check and escape newTitle
+    if (StringUtils.isBlank(filename)) {
+      throw new IllegalFileNameException("Provided filename is illegal");
+    }
+
+    Node node = nodeByUUID(config.getFileId(), config.getWorkspace());
+    Node parent = node.getParent();
+
+    String path = parent.getPath() + "/" + filename;
+    Session userSession = getUserSession(config.getWorkspace());
```

Comment:
You already have a Session in the `node` or `parent`, just use it `node.getSession()`

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/9/commits/4247d0da4c03177407f5d3ca4ff379ac414d2bbb